app/src/main/java/com/kms/cura/utils/GPSTracker.java

```diff
@@ -0,0 +1,156 @@
+package com.kms.cura.utils;
+
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+/**
+ * Created by linhtnvo on 6/13/2016.
+ */
+public class GPSTracker extends Service implements LocationListener {
+    private final Context context;
+
+    private boolean isGPSEnabled = false;
+    private boolean isNetworkEnabled = false;
+    private boolean canGetLocation = false;
+    private boolean error = false;
+
+
+    private Location location;
+
+    private double latitude;
+    private double longitude;
+    private String errorMessage = null;
+
+    private static final long MIN_DISTANCE_CHANGE_FOR_UPDATES = 10;
+    private static final long MIN_TIME_BW_UPDATES = 1000 * 60 * 1;
+
+    protected LocationManager locationManager;
+
+    public GPSTracker(Context context) {
+        this.context = context;
+        getLocation();
```

Comment:
when creating the GPSTracker, I make sure that I can get the location at that time so I call getLocation() in constructor